docs/the-new-architecture/pure-cxx-modules.md

```diff
@@ -0,0 +1,472 @@
+
+Finally, we have the constructor, that accepts a pointer to the `CallInvoker`, to communicate with JS if needed and the function's prototype we have to implement.
+
+The `NativeSampleModule.cpp` files is the actual implementation of our Turbo Native Module and implements the constructor and the method that we declared in the specs.
+
+## 4. Register the Module in the platform
+
+The next steps will let us register the module in the platform. This is the step that exposes the native code to JS so that the React Native application can finally call the native methods from the JS layer.
+
+This is the only time when we will have to write some platform-specific code.
+
+### Android
+
+To make sure that the iOS app can effectively build the C++ Turbo Native Module, we need to:
+
+1. Create a `CMakeLists.txt` to access our C++ code
+2. Modify `gradle.build` to point to the newly created `CMakeLists.txt` file.
+3. Create an `onLoad.cpp` file in our Android app to register the new Turbo Native Module.
+
+#### 1. Create the `CMakeList` file
+
+Android uses CMake to build. CMake needs to access the files we defined in our shared folder, to be able to build them.
+
+1. Create a new folder `SampleApp/android/app/src/main/jni`. The `jni` folder is where the C++ side of Android lives
+2. Create a `CMakeLists.txt` file and add this context
+
+```shell
+cmake_minimum_required(VERSION 3.13)
+
+# Define the library name here.
+project(appmodules)
+
+# This file includes all the necessary to let you build your application with the New Architecture.
+include(${REACT_ANDROID_DIR}/cmake-utils/ReactNative-application.cmake)
+
+# Define where the additional source code lives. We need to crawl back the jni, main, src, app, android folders
+target_sources(${CMAKE_PROJECT_NAME} PRIVATE ../../../../../shared/NativeSampleModule.cpp)
+
+# Define where CMake can find the additional header files. We need to crawl back the jni, main, src, app, android folders
+target_include_directories(${CMAKE_PROJECT_NAME} PUBLIC ../../../../../shared)
+```
+
+The CMake file does the following things:
+
+- Specifies the version of CMake we need to use and sets the `VERBOSE` flag.
+- Defines the library name to which our code will be added
+- Includes the base React Native's CMake file
+- Adds the source code that we need to build with the `target_sources` directives. You can see that we need to crawl back from the `jni` folder to the `shared` folder where our C++ TM lives.
+- Specify where CMake can find the additional header files. Also in this case we need to crawl back from the `jni` folder.
+
+#### 2. Modify gradle.build to link the C++ code
+
+Gradle is the tool that orchestrates the Android build. We need to tell it where it can find the `CMake` files to build the Turbo Native Module.
```

Comment:
```suggestion
Gradle is the tool that orchestrates the Android build. We need to tell it where it can find the `CMake` files to build the Turbo Native Module.
```